static/main.js

```diff
@@ -19,31 +19,68 @@ function bracketDevicePixelRatio() {
     return brackets[brackets.length - 1];
 }
 
+/**
+ * Finishes setting up the map
+ *
+ * @param {Object|null} config Config object
+ * @param string [config.attribution] Attribution text to show in footer; see below for default
+ * @param number [config.maxzoom=18] Maximum zoom level
+ */
+function setupMap( config ) {
+	var layerSettings,
+		query = '',
+		matchLang = location.search.match(/lang=([-_a-zA-Z]+)/);
+
+	if (matchLang) {
+		query = '?lang=' + matchLang[1];
+	}
+
+	config = config || {};
+
+	layerSettings = {
+		maxZoom: config.maxzoom ? config.maxzoom : 18,
+
+		// TODO: This is UI text, and needs to be translatable.
+		attribution: config.attribution ? config.attribution : 'Map data &copy; <a href="http://openstreetmap.org/copyright">OpenStreetMap contributors</a>',
```

Comment:
How about  `config.attribution || 'Map data &copy; <a href="http://openstreetmap.org/copyright">OpenStreetMap contributors</a>'` -- seems to be cleaner?  Same for maxzoom. This is a fairly common pattern in JS.

Reply:
Also, we are assuming that 'attribution' cannot be an empty string? I am not against it, but we do need to acknowledge that. If we want to allow empty string, we should do something like 

`attribution: config.attribution !== undefined ? config.attribution : 'default value goes here';`

or something like that. It might be better to **not** allow for an empty attribution if we're sure all sources are from OpenStreetMap, but if anyone uses this library to pull from anything else and doesn't want to put attribution (and it's legal) then empty string **may** be valid. Just raising this for consideration. @nyurik what do you think?

Reply:
All good points, but I can't do both (unless I use different loops for maxzoom and attribution).

We can't use Object.assign yet (IE 10 and 11).  It's probably better not to anyway, due to the 'specific config options' issue you mention (this is all public data, but Leaflet may not interpret field names the same way; there is a [leaflet-tilejson](https://github.com/kartena/leaflet-tilejson), but we're not using it).

I believe all of the WMF tiles use OSM, which [requires attribution](https://www.openstreetmap.org/copyright).

However, there are public domain sources, e.g. [Natural Earth](http://www.naturalearthdata.com/), so in principle some user of this code (and different tiles) might not want attribution.  This is an edge case, but I went with your suggestion.  That means I can't use `||`.